src/prommis/uky/costing/custom_costing_example.py

```diff
@@ -0,0 +1,171 @@
+    # Register custom currency units used in the custom costing model
+    pyunits.load_definitions_from_strings(
+        [
+            "USD_custom = 500/700 * USD_CE500",
+        ]
+    )
+
+    # define global parameters - base cost year and period
+    def build_global_params(self):
+
+        # Set the base year for all costs
+        self.base_currency = pyunits.USD_2021
+        # Set a base period for all operating costs
+        self.base_period = pyunits.year
+
+    # custom costing method
+    def cost_custom_vessel(
+        blk,  # when the costing block is built, blk will be the costing block itself
+        volume_per_unit=1000 * pyunits.m**3,
+        material="carbonsteel",  #
+        water_injection_rate_per_unit=1 * pyunits.m**3 / pyunits.s,
+        number_of_units=1,
+        CE_index_year="2021",
+    ):
+
+        # alias for cost year units
+        CE_index_units = getattr(pyunits, "MUSD_" + CE_index_year)
+
+        # make parameter for number of units
+        blk.number_of_units = Param(initialize=number_of_units, mutable=False)
+
+        # define the bare erected cost per unit
+
+        material_factor_dict = (
+            {  # material factors for each valid choice of shell material
+                "carbonsteel": 1,
+                "stainlessteel": 1.5,
+                "aluminum": 2,
+            }
+        )
+
+        blk.capital_cost_per_unit = Var(
+            initialize=1000,
+            units=CE_index_units,
+            domain=NonNegativeReals,
+            bounds=(0, None),
+        )
+
+        @blk.Constraint()
+        def capital_cost_per_unit_eq(blk):
+            # cost equation CAPITAL_COST = REF_COST * (VOLUME / REF_VOLUME)**0.6
+            ref_cost = 10000 * pyunits.USD_custom
+            ref_volume = 1000 * pyunits.m**3
+            return blk.capital_cost_per_unit == pyunits.convert(
+                material_factor_dict[material]
+                * ref_cost
+                * (volume_per_unit / ref_volume) ** 0.6,
+                to_units=CE_index_units,
+            )
+
+        # create a variable capital_cost that the REE Costing Framework can look for
+        blk.capital_cost = Var(
+            initialize=1000,
+            units=CE_index_units,
+            domain=NonNegativeReals,
+            bounds=(0, None),
+        )
+
+        @blk.Constraint()
+        def capital_cost_constraint(blk):
+            return blk.capital_cost == blk.capital_cost_per_unit * blk.number_of_units
+
+        # define the fixed costs
+
+        blk.fixed_operating_cost_per_unit = Var(
+            initialize=1000,
+            units=CE_index_units,
+            domain=NonNegativeReals,
+            bounds=(0, None),
+        )
+
+        @blk.Constraint()
+        def fixed_operating_cost_per_unit_eq(blk):
+            return blk.fixed_operating_cost_per_unit == pyunits.convert(
+                0.05 * blk.capital_cost, to_units=CE_index_units
+            )
```

Comment:
Is it just assumed that the fixed operating cost is 5% of the unit's capital cost? Is it worth making the 5% a parameter or is this assumption always the go-to?

Reply:
I hard-coded the value just for this example as a way to test the functionality. Normally the CAPEX, fixed OPEX, and variable OPEX equations would come from surrogates or a bottom-up modeling approach when a user or developer is writing a custom cost model.